packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -107,6 +112,23 @@ library LibProposing {
         blk.verifiedForkChoiceId = 0;
         blk.metaHash = LibUtils.hashMetadata(meta);
         blk.proposer = msg.sender;
+        blk.gasLimit = input.gasLimit;
+
+        // Let's see if calculations are correct so first just
+        // debug them. (Curly braces due to stack too deep error)
+        console2.log("----------------------------------------------------");
+        console2.log("------------------In proposeBlock()-----------------");
+        {
+            // Need to scale 'down', because baseFeeProof is scaled 'up'
+            // Todo: Could you please Brecht advise on the math ?
+            // So since in baseFee() function i needed to scale up, otherwise divison with zero
+            // somewhere we need to scale down..
+            uint256 feeNew = (input.gasLimit *
+                state.baseFeeProof); /*  / SCALING_FACTOR; (??) */
```

Comment:
Ideally it's not necessary to do any scaling here, but if the taiko token is used here and that one now only has 8 decimals it may be needed unfortunately.